test/Microsoft.AspNet.Mvc.Core.Test/HttpNotFoundObjectResultTest.cs

```diff
@@ -46,80 +41,43 @@ public void HttpNotFoundObjectResult_InitializesStatusCodeAndResponseContent()
         public async Task HttpNotFoundObjectResult_ExecuteSuccessful()
         {
             // Arrange
-            var input = "Test Content";
-            var stream = new MemoryStream();
-
-            var httpResponse = new Mock<HttpResponse>();
-            var tempContentType = string.Empty;
-            httpResponse.SetupProperty(o => o.ContentType);
-            httpResponse.SetupGet(r => r.Body).Returns(stream);
-
-            var actionContext = CreateMockActionContext(httpResponse.Object);
-            var notFound = new HttpNotFoundObjectResult(input);
+            var httpContext = GetHttpContext();
+            var actionContext = new ActionContext()
+            {
+                HttpContext = httpContext,
+            };
 
+            var result = new HttpNotFoundObjectResult("Test Content");
+            
             // Act
-            await notFound.ExecuteResultAsync(actionContext);
+            await result.ExecuteResultAsync(actionContext);
 
             // Assert
-            httpResponse.VerifySet(r => r.StatusCode = StatusCodes.Status404NotFound);
-            Assert.Equal(input.Length, httpResponse.Object.Body.Length);
+            Assert.Equal(StatusCodes.Status404NotFound, httpContext.Response.StatusCode);
         }
 
-        private static ActionContext CreateMockActionContext(
-            HttpResponse response = null,
-            string requestAcceptHeader = "application/*",
```

Comment:
There was a big ball of Moq-mud that was copy-pasted all over these tests. I've simplified it substantially.